Every outfit and ship now has tags that missions can use

Elyssaen opened the pull request:
I've added to every outfit and ship in the game a series of tags that look like this:

```
outfit "Surveillance Pod"
	category "Systems"
	tags "cargo scanner" "outfit scanner"
	cost 42000
```

You can condition missions on them like this:

```
	to offer
		has "tag: cargo scanner"

		# Alternatively:
		"tag: cargo scanner" >= 2
```

Here are the advantages:
1. Your mission requires a cargo scanner? `require outfit` doesn't let players a Surveillance Pod – tags do.
2. Your mission requires a specific number of some outfit? `require outfit` can't help you – tags can.
3. You have a ship with an intrinsic attribute (like a cloak) that's messing up your mission requiring a cloaking device? Ships have these tags too.
4. You want to make a mission that notices you've stolen Quarg technology, or you're using Lovelace weapons? You can do this with tags.
5. You want a mission where you have to show up unarmed? Weapons are tagged as such, so go wild.
6. Want to either look at what's on your flagship, or what's in your _fleet_? Got you covered: use "fleet tag: X" to count items tagged X in the player's entire fleet, and "tag: X" to just look at their flagship.
7. Want to offer a mission based on the number of warships in your fleet, not differentiating by Light/Medium/Heavy? I've added `warship` tags that will let you do this.

<details>
<summary>Technical implementation notes</summary>
1. They're brought in as attributes prefaced by "tag: ", so they use all the same code as attributes.<br />
2. When a specific ship is saved to file, they're turned back into tags.<br />
3. They're turned into mission-friendly conditions using PlayerInfo::UpdateAutoConditions, the same function that updates `cargo space`, `passenger space` and e.g. `ships: Light Warship`.<br />
</details>

ReimeiSky commented on data/deprecated outfits.txt:
```diff
@@ -121,6 +127,7 @@ outfit "Mass Expansion"
 # Deprecated to give it a less generic name.
 outfit "Reverse Thruster"
 	category "Engines"
+	tags human
```
https://github.com/Elyssaen/endless-sky/pull/11/files#diff-9ee530738ea1b89dcd0bb31318abade4R519 k

Elyssaen replied:
You're going to say it should be "deep sky" as well, aren't you? I missed that. Will update! 

Incidentally, I _knew_ you of everyone here would be the guy to help me catch errors in tags. : )

Just a note, I noticed some things, like Armageddon Cores, are only sold at one outfitter (Lovelace Advanced), but there wasn't any in-game justification for Lovelace manufacturing reactors. I figure it's possible it's just meant as a shortcut to get them to appear only in advanced outfitters in certain places, and doesn't necessarily mean they should be tagged Lovelace. So I've kept manufacturer tags mostly to obvious things. Another thing is not marking the Hauler III as Southbound, even though they're the only ones selling it, as I can't find any indication the Hauler line is necessarily theirs.

Elyssaen committed:
Implemented outfit and ship tags

Any outfit or ship can have a series of tags, e.g. tags alien remnant
warship, and each tag is turned into auto-conditions that can be used
for mission logic. This can allow, for instance, a mission that triggers
if you have Quarg technology, a sponsorship deal if you use Lovelace
weapons, or a mission requiring you have the ability to scan cargo
(instead of specifically the Cargo Scanner outfit).

Elyssaen committed:
Added tags to every outfit and ship

Elyssaen commented:
Finally opened https://github.com/endless-sky/endless-sky/pull/2617.
